frontend/src/store/store.js

```diff
@@ -30,15 +30,16 @@ export const mutations = {
 }
 
 export const actions = {
-  login: ({ dispatch, commit }, token) => {
-    const decoded = VueJwtDecode.decode(token)
-    commit('token', token)
-    commit('email', decoded.email)
-    commit('language', decoded.language)
-    commit('username', decoded.username)
-    commit('firstName', decoded.firstName)
-    commit('lastName', decoded.lastName)
-    commit('description', decoded.description)
+  login: ({ dispatch, commit }, data) => {
+    // const decoded = VueJwtDecode.decode(data.token)
+    const { user } = data
+    commit('token', data.token)
```

Comment:
Here we have to clarify if this is not too asynchronous and the commit can last to long so calls will fail because of a missing or invalid token?!

Reply:
I cannot see a problem here. The client stores the token received and never knows if it is valid or not.

Reply:
fine fine. I just think that `commit` takes some time and this could prevent a quick call to succeed.